### ISSUE TITLE
Add support for private Kubernetes clusters

### ISSUE THREAD

**Issue by Stono** [terraform-providers/terraform-provider-google#1174]
Hi there,
With Kubernetes 1.9.2 on GKE (see release notes: https://groups.google.com/forum/#!topic/google-cloud-sdk-announce/GGW3SQSANIc) there is now a private cluster flag, allowing us to deploy internal ranges for the master:

```
Added --private-cluster flag to enable PrivateCluster creation:
        gcloud {alpha,beta} container clusters {create} --private-cluster
        --master-ipv4-cidr <CIDR>
```

Would it be possible to add this to [google_container_cluster](https://www.terraform.io/docs/providers/google/r/container_cluster.html) ?

**Comment by darrenhaken** [comment]
I'll pick this up

**Comment by Stono** [comment]
@danawillow, @darrenhaken won't be picking this up - it was an error and he wrote it on the wrong thread because i linked him the wrong issue :-)

**Comment by lenartj** [comment]
Funny it happened like that, I am actually interested in picking this one up as it is needed for a project.

**Comment by Stono** [comment]
Please do!

**Comment by lenartj** [comment]
Just an update. I have a working prototype of this change. Tomorrow I will start adding some PRs and link them to this issue.

- [x] Proof of Concept
- [ ] Submit PR to update vendor
- [ ] Fix the tests
- [ ] Add some sanity checks
- [ ] Submit PR to add `private_cluster` and `master_ipv4_cidr`

**Comment by Stono** [comment]
Ah awesome.  Unfortunately I'm away on holiday for a week but I'll be sure
to test it out when I get back.

Thanks so much for your efforts
Karl

**Comment by danawillow** [comment]
Fixed in #1250
